mechanics/Release-Mechanics.md

```diff
@@ -0,0 +1,45 @@
+# Release Mechanics
+
+Knative core components (repo) releases from master are coordinated for the
+following reasons:
+
+1. Coordinated release makes it easier to release documentation and other
+   cross-cutting repositories.
+1. Coordinated release makes it easier for operators to install and describe
+   what release they are running.
+1. Coordinated release makes it easier to test compatibility across repos,
+   though cross-repo dependencies should generally be supported at least +/- 1
+   revision (i.e. revision N+1 of repo X should work with revision N of repo Y
+   and vice-versa).
+
+**Patch (cherrypick) releases are not coordinated across repos**; it is expected
+that each repo will perform patch releases for critical bugs and security issues
+for the last 2 releases.
+
+## Release Schedule
+
+Knative releases will be cut (automatically) every 6 weeks. Any change to this
+schedule should be coordinated by the TOC.
+
+<!-- TODO: do we have a calendar for this? What about releases during holidays like Christmas? -->
+
+## Release Naming
+
+Regular releases from master will be named by the date of the release in ISO
+date format (e.g. `2019-08-06`). Patch releases will be named as `2019-08-06.1`,
+`2019-08-06.2`, etc. The head release will be named without a ".0" suffix; the
+first patch (cherrypick) release should be named ".1", with each subsequent
+patch incrementing the number.
+
+Each repo should define a mapping from the canonical date-based format to a
```

Comment:
Overall looks good. It will be helpful if we standardize how this is defined (or exposed to end-users) such that an end-user has a consistent way of figuring out mapping from date --> semver for all knative repos.

Reply:
Do you have a proposal for where this should go?

I wanted to get a base-line recommendation in here and then iterate on the mapping documentation based on feedback from docs/WG leads.

Reply:
thinking more about it there I think we need to clarify more in this doc. (it could also be my lack of knowledge, so please point me to docs if that is the case :))
 -  What does release naming mean?
Today when we release there are multiple places that we mention the semver:
 - release branch name
 - tag in github
 - labels on k8s artifacts
 - tag in the container images (our docs recommend customers to see this to figure out the version they have installed)
 - docs page dropdown.

semver carries a sematic meaning so I believe we want to continue with it.
However with each component updating semver differently (example eventing v0.10, serving v1.0), we want to tie them together with a date based version.

So the question is what that date based version exactly mean and where do users see it? Clarifying that in this doc is important.
One option could be - let each repo continue what they are doing right now.
In docs have a page that maintains the mapping of date-semver, and each repo's release notes maintains it for its own release.
We could also have both date and semver in the version labels and tags. We can figure this out and come up with proposals.

However, clarifying exactly where the date based version or Release Name will appear and how users will see it is needed in this doc. Then we can come up with a scheme for the mapping to semver. Makes sense?

Reply:
I would use the date-based names in all of the above automated locations.

I might include the semantic version on the landing page for the component, ala https://knative.dev/docs/serving might say:

# Knative Serving (v1.1)

Reply:
That sounds good to me. Along with the landing page semantic versioning could also be mentioned with release notes in Github (along with the date-based version).
About date-based v/s just a number, I would prefer date-based as it conveys an extra bit of information about the release date along with the sequential info.

Reply:
What semVer would a patch release have?